src/sidebar/services/view-filter.js

```diff
@@ -183,7 +182,9 @@ function viewFilter(unicode) {
     const rootFilter = new BinaryOpFilter('and', fieldFilters);
 
     return annotations
-      .filter(ann => rootFilter.matches(ann))
+      .filter(ann => {
+        return rootFilter.matches(ann) && ann.id;
```

Comment:
An alternative way of resolving the issue would be to swap the order of the checks here (`return ann.id && rootFilter.matches(ann)`) so that the logic short-circuits and the filters never have to deal with the possibility of a non-saved annotation.